03_cross_cancer_classification/nbconverted/p53_stability_pilot.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## p53 feature stability analysis
+# 
+# How stable (in terms of performance, and eventually coefficients) are TP53 mutation prediction models fit on the same dataset? We're particularly interested in how they generalize across domains (in our case tissues/cancer types): can models with similar performance on the training set have very different generalization performance?
+
+# In[1]:
+
+
+import os
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+
+# ### Load results
+
+# In[2]:
+
+
+results_dir = os.path.join(
+    cfg.repo_root, '03_cross_cancer_classification', 'results'
+)
+p53_results_dir = os.path.join(
+    results_dir, 'p53_stability_pilot', 'cross_cancer'
+)
+
+
+# In[3]:
+
+
+cross_cancer_df = au.load_prediction_results_cc(p53_results_dir, 'cross_cancer')
+print(cross_cancer_df.shape)
+cross_cancer_df.head()
+
+
+# ### Look at overall cross-cancer performance
+
+# In[4]:
+
+
+heatmap_df, sorted_ids = au.heatmap_from_results(cross_cancer_df,
+                                                 normalize_control=True,
+                                                 sort_results=False)
+heatmap_df.iloc[:5, :5]
+
+
+# In[5]:
+
+
+sns.set({'figure.figsize': (15, 8)})
+sns.heatmap(heatmap_df, cbar_kws={'label': 'AUPR difference from baseline'}, center=0)
+plt.title('Cross-cancer mutation detection, AUPR heatmap')
+plt.xlabel('Test identifier')
```

Comment:
Is there something about UCEC that makes it easier to predict? It seems to have overall better performance. Maybe I'm reading too much into the heatmap.

Reply:
I'm not sure! I'm not surprised that SARC and LGG (the two non-carcinoma cancer types) are generally harder to predict than the others, but I'm not sure why UCEC would be easier to predict than the rest of the carcinomas.

Maybe it's that the labels are a bit more imbalanced (around 35% mutated as opposed to closer to 50% for most of the other cancer types)? BRCA has around the same % mutated as UCEC, though, and performance isn't as good.